db/model/subject.js

```diff
@@ -345,8 +349,32 @@ module.exports = function subject(seq, dataTypes) {
           }
         }
 
+        const subjectDepth =
+             inst.absolutePath.split(constants.absolutePathSeparator).length;
+
+        /*
+         * If the depth of the subject does not match the hierarchyLevel
+         * throw an error
+         */
+        if (subjectDepth !== inst.hierarchyLevel) {
+          throw new InvalidHierarchyLevel('The updated depth of ' +
+            `${inst.absolutePath}: ${subjectDepth}, does not match the ` +
+             `updated hierarchyLevel: ${inst.hierarchyLevel}`);
+        }
+
```

Comment:
@iamigo @pallavi2209 This check fails in Travis wherever I put this. I also tried to write a sequelize raw query to throw an error when a root subject has an entry in the Subjectsancestors table. This validation does not run locally. Any other suggestions on how to handle it? Also, I have a validation on the API layer to prevent any changes to the hierarchyLevel field, with the API level validation in place, do we need the DB validation too?